lib/Conversion/TritonGPUToLLVM/ViewOpToLLVM.cpp

```diff
@@ -39,6 +39,21 @@ struct SplatOpConversion : public ConvertOpToLLVMPattern<triton::SplatOp> {
       for (unsigned i = 0; i < ratio; ++i)
         vec = insert_element(vecType, vec, intCst, int_val(32, i));
       constVal = vec;
+    } else if (mlir::VectorType srcElemVecTy = srcType.dyn_cast_or_null<mlir::VectorType>()) {
```

Comment:
srcType shouldn't be nullptr so you don't need `dyn_cast_or_null` (should be `dyn_cast`). In what case would the converted type be a vector? I though we were always generating scalars, I'm wondering when we would get such type.
This probably requires a lit test to be properly tested.

Reply:
Thanks for the comments. I changed to use `dyn_cast`. 

For your question, the dot_op for mfma use a vector type as here: https://github.com/openai/triton/blob/441d6b10148e313552192ebbd1fbd30d5d93b563/third_party/amd/lib/TritonAMDGPUToLLVM/ConvertLayoutOpToLLVM/SharedToDotOperandMFMA.cpp#L539.

Reply:
Interesting, I wonder if this is going to be a problem for other lowering, I believe we have quite a few places where we expect scalars.

Reply:
Could you please point us one place, so we can get some idea about that?
If it really causes many problems, maybe we can consider refactoring not to use the vectype. 
cc: @jayfurmanek, @zhanglx13, @binarman

Reply:
@ThomasRaoux 
Here is the place where the typeconverter returns a vec type
https://github.com/openai/triton/blob/dd8c270e5271523b9672eaa515117e6f2f5f0be6/lib/Conversion/TritonGPUToLLVM/TypeConverter.cpp#L85-L87
For dotOperand whose parent is mfma, we group the elements per thread per mfma instruction into a vector. I don't have a strong argument why this is necessary or beneficial. But if you only want to handle scalar as constants, I think we can figure out a way to handle scalars as well.

Reply:
> Interesting, I wonder if this is going to be a problem for other lowering, I believe we have quite a few places where we expect scalars.

The vector should only be generated on AMD path when mfma is used. So nothing should break on the NV path.

Reply:
yeah I'm mainly wondering if we will find a bunch of problems in common patterns when used with mfma layout. Wouldn't this be a problem for elementwise patterns for instance? Any pattern that doesn't support vector would fail?

Reply:
I agree that the current impl is not future-proof. And it should be better for the type converter to return the same type for dotOperand layout.
@scxiao  Let's sync offline to figure out a way to fix the test from a different direction

Reply:
> yeah I'm mainly wondering if we will find a bunch of problems in common patterns when used with mfma layout. Wouldn't this be a problem for elementwise patterns for instance? Any pattern that doesn't support vector would fail?

Yes, I remember we met the problem when converting fp8 to fp16 for the mfma dot_op and we add a special processing for that.

Reply:
> I agree that the current impl is not future-proof. And it should be better for the type converter to return the same type for dotOperand layout. @scxiao Let's sync offline to figure out a way to fix the test from a different direction

OK, we can talk offline for that.